matching/views.py

```diff
@@ -23,9 +23,14 @@ def show_potential_matches(request, utrip_id):
         messages.error(request, "Please select a valid trip")
         return redirect(reverse("trip:view_trips"))
 
-    # applying dynamic filter to generate match pool
+    # applying dynamic filter to generate match pool where
+    # travel_type = companion AND
+    # destination is matching AND
+    # trip dates are overlapping (for at least 1 day) AND
+    # user's age is in defined range
     matching_trips = UserTrip.objects.filter(
-        start_trip__lt=current_usertrip.end_trip,
```

Comment:
One other thing: since we have the ability to milestone trips, which is controlled by a UserTrip attribute is_active (see UserTrip in trip/models.py and milestone_trip() in trip/views.py), we should be filtering trips that have been milestoned out of this queryset.

We can add another filter to matching_trips like is_active=True. 

If we don't filter this, users will still appear in the match structure after they've milestoned their trip

Reply:
Good point, will address this one.

Also, would it be possible that a trip remains active, but the user has deactivated their profile? So, even if we sort of send the matching request to the user, but the possibility of it being accepted is very less?

Reply:
So we do have on_delete = models.CASCADE set up in the UserTrip model, but I'm not sure whether Django handles a change to is_active in the User model as a deletion or not. I think it might, since by default you can't log in if is_active is set to false, but we'll have to test that out (create an account, create a trip for them, then delete the account and see if it still shows up in another account's match set) and handle it if it still shows up.

Reply:
The `is_active` flag being set to `False` does not trigger the CASCADE behavior unless you explicitly handle it in your code. Setting `is_active` to `False` is typically a way to mark an object as inactive or "soft deleted" without actually removing it from the database. It is a common practice for implementing soft deletion.

Read it on an article.

Reply:
Hmm okay. So we'll have to check whether usertrip's user's active flag is set to true when we filter then.